src/cache/client.ml

```diff
@@ -45,20 +45,37 @@ let make ?finally ?duplication_mode ~command_handler () =
     and finally stdout = ignore (Unix.close_process_in stdout) (* FIXME *) in
     Exn.protectx (Unix.open_process_in cmd) ~finally ~f
   in
-  let* addr, port =
-    match String.split_on_char ~sep:':' port with
-    | [ addr; port ] -> (
-      match Int.of_string port with
-      | Some i -> (
-        try Result.Ok (Unix.inet_addr_of_string addr, i)
-        with Failure _ ->
-          Result.Error (errf [ Pp.textf "invalid address: %s" addr ]) )
-      | None -> Result.Error (errf [ Pp.textf "invalid port: %s" port ]) )
-    | _ -> Result.Error (errf [ Pp.textf "invalid endpoint: %s" port ])
-  in
-  let fd = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-  let* _ =
-    Result.try_with (fun () -> Unix.connect fd (Unix.ADDR_INET (addr, port)))
+  let* fd =
+    match
+      ( String.drop_prefix ~prefix:"unix:" endpoint
```

Comment:
Perhaps this should be `unix://` for consistency?

Reply:
`//` in a uri introduces a hostname; I think we do want `unix:/some/path`, which is scheme `unix` and path `/some/path`.